### PR TITLE
Add plumber_options()

### PR DESCRIPTION
**Pull request by meztez**
PR task list:
- [x] Update NEWS 
- [x] Add tests
- [x] Update documentation with `devtools::document()`

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -42,6 +42,8 @@ plumber 0.5.0
 
 ### New features
 
+* Document plumber options. Add `setPlumberOptions` and `getPlumberOptions`. (@meztez)
```
Comment (schloerke) [inline NEWS.md:45]:
```suggestion
* Document plumber options. Add `setPlumberOptions` and `getPlumberOptions`. (@meztez, #555)
```

**Comment by schloerke** [comment]
@meztez May we combine these functions into a single function? `optionsPlumber()`

This will follow pattern of `options` (as well as autocomplete) and not take away from `plumber*` methods.

File: R/plumber-options.R [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+# for (f in dir("R", full.names = T)) {
+#   ct <- paste0(readLines(f, warn = F), collapse = "")
+#   m <- stringi::stri_match_all_regex(ct, "getOption\\([^,\\)]+,?\\)?")[[1]][,1]
+#   m <- gsub("\\s", "", m)
+#   if (length(m) > 0 && !is.na(m)) {
+#     mm <- c(mm, m)
+#   }
+# }
+# mm <- unique(sort(gsub("getOption|\\(|\"|,|'|\\)", "", mm)))
+# mm <- paste0("#' \\item{", mm, " (defaults to `fill`)}{document})")
```
Comment (schloerke) [inline R/plumber-options.R:12]: Can we add a testthat test for this instead?

Untested code below:

```r
test_that("all options used are optionsPlumber parameters", {
  skip_on_cran() # only test in CI / locally

  ## `./R` will not exist when shiny is installed
  r_folder <- "../../R"
  if (
    # not local structure
    !dir.exists(r_folder) || 
    # must contain many files, not just the three files typically found in installed folder
    length(dir(r_folder)) < 10
  ) {
    skip("Not testing locally. Skipping")
  }

  matches <- character()
  for (r_file in dir(r_folder, full.names = T)) {
    file_content <- paste0(readLines(r_file, warn = F), collapse = "")
    match <- stringi::stri_match_all_regex(file_content, "getOption\\([^,\\)]+,?\\)?")[[1]][,1]
    match <- gsub("\\s", "", match)
    if (length(match) > 0 && !is.na(match)) {
      matches <- c(matches, match)
    }
  }
  options_used <- unique(sort(gsub("getOption|\\(|\"|,|'|\\)", "", matches)))

  ### code to match formals
  options_plumber_formals <- paste0("plumber.", sort(names(formals(optionsPlumber))))
  
  expect_equal(
    options_used, 
    options_plumber_formals
  )
})
```

File: R/plumber-options.R [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+#' @param maxRequestSize see \code{\link{plumber-options}}
+#' @param postBody see \code{\link{plumber-options}}
+#' @param port see \code{\link{plumber-options}}
+#' @param sharedSecret see \code{\link{plumber-options}}
+#' @param swagger.url see \code{\link{plumber-options}}
+#' @details
+#' Sets plumber options. Call without arguments to get current
+#' values.
+#' @export
+optionsPlumber <- function(
```
Comment (cpsievert) [inline R/plumber-options.R:44]:
```suggestion
options_plumber <- function(
```

Comment (cpsievert) [reply R/plumber-options.R:44]: Let's make all new, independent, functions snake_case

File: R/plumber-options.R [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    missing(postBody),
+    missing(port),
+    missing(sharedSecret),
+    missing(swagger.url)
+    )) {
+    options_names <- grep("^plumber", names(options()), value = TRUE)
+    set_options <- lapply(options_names, getOption)
+    names(set_options) <- options_names
+    return(set_options)
+  }
```
Comment (cpsievert) [inline R/plumber-options.R:66]: If no values are provided, the `options()` call below, still returns a named list of the current option values

```suggestion
```

File: R/plumber-options.R [hunk 5]
```diff
@@ -0,0 +1,76 @@
+
+#' How to use Plumber options
+#'
+#' @section Options:
+#' There are a number of global options that affect Plumber's behavior. These can
+#' be set globally with \code{\link{options}} or with \code{\link{optionsPlumber}}. Current values
+#' can be retrieved with \code{\link{getOption}} or \code{\link{optionsPlumber}} without arguments.
```
Comment (cpsievert) [inline R/plumber-options.R:7]: Use markdown links (e.g. `[getOption()]`)

File: tests/testthat/test-options.R [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+test_that("Options set and get", {
+  option_value <- getOption("plumber.postBody")
+  options_plumber(postBody = FALSE)
+  expect_false(getOption("plumber.postBody"))
+  options_plumber(postBody = NULL)
+  expect_null(getOption("plumber.postBody"))
+  options(plumber.postBody = option_value)
+})
+
+test_that("all options used are optionsPlumber parameters", {
```
Comment (schloerke) [inline tests/testthat/test-options.R:12]:
```suggestion
test_that("all options used are `options_plumber()` parameters", {
```

File: NEWS.md [hunk 7]
```diff
@@ -48,6 +48,8 @@ plumber 0.5.0
 
 ### New features
 
+* Document plumber options. Add `optionsPlumber`. (@meztez, #555)
```
Comment (schloerke) [inline NEWS.md:51]:
```suggestion
* Document plumber options. Add `options_plumber`. (@meztez, #555)
```

**Pull request merged**